realm/realm-library/src/objectServer/java/io/realm/internal/util/BsonConverter.java

```diff
@@ -0,0 +1,224 @@
+     *
+     * Converts primitive boxed types to the equivalent BSON equivalent value object and {@link List}
+     * of values into {@link BsonArray} of converted values.
+     *
+     * {@link BsonValue} objects are left as is.
+     *
+     * @param value The object to convert.
+     * @return BSON value representation of the origin value object.
+     *
+     * @throws IllegalArgumentException If the object could not be mapped to a BSON type.
+     */
+    // FIXME Review supported types...any obvious types missing?
+    public static BsonValue to(Object value) {
+        if (value instanceof BsonValue) {
+            return (BsonValue) value;
+        }
+        // Convert list to BsonArray
+        else if (value instanceof List) {
+            return BsonConverter.to(((List) value).toArray());
+        }
+        // Native types
+        else if (value instanceof Integer) {
+            return new BsonInt32((Integer) value);
+        } else if (value instanceof Long) {
+            return new BsonInt64((Long) value);
+        } else if (value instanceof Boolean) {
+            return new BsonBoolean((Boolean) value);
+        } else if (value instanceof String){
+            return new BsonString((String) value);
+        } else if (value instanceof byte[]) {
+            return new BsonBinary((byte[]) value);
+        }
+        // Bson values
+        else if (value instanceof ObjectId) {
+            return new BsonObjectId((ObjectId) value);
+        }
+        else if (value instanceof Decimal128) {
+            return new BsonDecimal128((Decimal128) value);
+        }
+        // FIXME Missing Realm types
+        // Date
+        // Float
+        // Double
+        // Object
+        // List
+        // LinkingObject
+        // FIXME Missing Bson value
+        throw new IllegalArgumentException("Conversion to BSON value not supported for " + value.getClass().getName());
+    }
+
+    /**
+     * Converts a list of objects to BSON values.
+     *
+     * @param value List of value objects to convert.
+     * @return A list of BSON values of the converted input arguments.
+     *
+     * @throws IllegalArgumentException If any of the value objects could not be converted to a
+     * BSON type.
+     *
+     * @see #to(Object)
+     */
+    public static BsonArray to(Object... value) {
+        ArrayList result = new ArrayList();
+        for (Object o1 : value) {
+            result.add(to(o1));
+        }
+        return new BsonArray(result);
+    }
+
+    /**
+     *
+     *
+     * @param clz
+     * @param value
+     * @param <T>
+     * @return
+     *
+     */
+    // FIXME Would we rather return null? Would maybe make it cleaner to use with
+    //  functions.callFunctionTyped("sum", BsonString::class.java, "Realm")?.value
+    //  but would also silently hide if expectations of type is wrong
+    private static <T extends BsonValue> T fromToBson(Class<T> clz, BsonValue value) {
```

Comment:
Not sure I understand the naming of this method?

Reply:
No. Leftover from before the interface settled. Removing it completely, as `from` now handles it.